slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -221,18 +228,25 @@ ::grpc::Status SLAMServiceImpl::GetCurrentPointCloudMap(
     try {
         std::shared_lock optimization_lock{optimization_shared_mutex,
                                            std::defer_lock};
-        if (optimization_lock.try_lock()) {
+        if ((optimization_lock.try_lock()) && (!localization_map_ready)) {
```

Comment:
[minor]
```suggestion
        if (action_mode != ActionMode::LOCALIZING && optimization_lock.try_lock()) {
```

Reply:
was already making changes so I added this to my commit